feature_engine/creation/decision_tree_features.py

```diff
@@ -0,0 +1,422 @@
+            # (2) the list is only comprised of integers.
+            if (
+                    max(strategy) > len(variables)
+                    or not all(isinstance(feature, int) for feature in strategy)
+            ):
+                raise ValueError(
+                    "strategy must be a list of integers and the maximum value cannot "
+                    "be greater than the number of variables to combine. "
+                )
+
+        if isinstance(strategy, tuple) and variables is not None:
+
+            # Check that the tuples only contains strings with variable names.
+            if (
+                not all(
+                isinstance(element, (str, int)) and element in variables
+                for tuples in strategy
+                for element in tuples
+                )
+            ):
+                raise ValueError(
+                    "If strategy is tuples, the elements of the tuples must be strings"
+                    "or integers present in the variable list entered in `variables`."
+                )
+        elif isinstance(strategy, tuple) and variables is None:
+            raise ValueError(
+                "if strategy is a tuple with variable names, `variables` cannot be "
+                "None."
+            )
+
+        if not isinstance(drop_original, bool):
+            raise TypeError(
+                "drop_original takes only boolean values True and False. "
+                f"Got {drop_original} instead."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.output_features = strategy
+        self.drop_original = drop_original
+        self.cv = cv
+        self.scoring = scoring
+        self.regression = regression
+        self.variables = _check_input_parameter_variables(variables)
+        self.param_grid = param_grid
+        self.random_state = random_state
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Fits decision trees with the provided the feature combinations.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just
+            the variables to transform.
+
+        y: pandas Series or np.array = [n_samples,]
+            The target variable that is used to train the decision tree.
+        """
+
+        # only numerical variables
+        self.variables_ = _find_or_check_numerical_variables(X, self.variables)
+
+        # basic checks
+        X, y = check_X_y(X, y)
+        _check_contains_na(X, self.variables_)
+        _check_contains_inf(X, self.variables_)
+
+        # get all sets of variables that will be used to create new features
+        self.variable_combinations_ = self._create_variable_combinations()
+        self._estimators = {}
+
+        # fit a decision tree for each set of variables
+        for combo in self.variable_combinations_:
```

Comment:
I think here you had a list first and with that you made a dictionary. I think we can put the 2 things together. This may have broken something. I apologize if that was the case.